<a href="https://colab.research.google.com/github/Umanggoel2/AirBnb-project/blob/main/Capstone_Project_AirBnb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name** -



##### **Project Type**    - Exploratory Data Analysis
##### **Contribution**    - Individual
##### **Member Name -** Umang Goel

# **Project Summary -**

***Introduction of AirBnb:***
* The San Francisco-based startup Airbnb was founded in 2007 when two hosts opened their residence to three visitors. Since then, the firm has expanded to over 4 million hosts and hosted over 1.5 billion guests in practically every nation.
* Airbnb serves as an online accommodation marketplace that links customers searching for short-term lodging with those wishing to lease out their properties. Airbnb gives hosts a comparatively simple method to make a little money from their homes.
* The basic premise of Airbnb is to connect locals with those who have extra space in their homes or apartments that they can rent out to tourists. With the security of knowing that a large firm will manage payments and provide other assistance, hosts utilizing the platform get to market their rentals to millions of individuals around the world.

***Purpose of EDA:***
* The main goal of this analysis is to understand the pattern and to provide useful insights into AirBnb to improve its functionality and serviceability.
* Firstly data is to be explored and cleaned using different data wrangling techniques. The missing or null values are to be handled carefully as they will create problems during data analysis and can affect statistics and computations. Duplicates and outliers are also dealt with in this cleaning process only as they can cause errors in data collection, data entry, or data merging and can lead to false analysis results.
* After cleaning of the data, the identification of different patterns and trends present inside the data can be done using various data visualization techniques. Various graphs and charts can be created to visualize the data, and record observations and insight which will be very useful for future analysis and decision-making related to Airbnb.

# **GitHub Link -**

Link- https://github.com/Umanggoel2/AirBnb-project

#**Problem Statement** -

Since 2008, guests and hosts have used Airbnb to expand on travelling possibilities and present a more unique, personalised way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analysed and used for security, business decisions, understanding of customers' and providers' (hosts) behaviour and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. This dataset has around 49,000 observations in it with 16 columns and it is a mix of categorical and numeric values. Explore and analyse the data to discover key understandings.

***Objectives-***

# ***Let's Begin !***

**Dataset Description**

The airbnb dataset contains information on various attributes of reservations, such as their room type, price, minimum nights, last review date and many more. This dataset includes a total of 48895 instances, with 16 different attributes.

###Importing the needed libraries:

In [174]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

###Dataset Loading:

In [175]:
path_name = "/content/Airbnb NYC 2019.csv"
airbnb_df = pd.read_csv (path_name)

Here, airbnb_df is a Pandas DataFrame object that contains the airbnb dataset.

### Dataset First View:

In [176]:
# Displaying complete dataset by calling the name
airbnb_df

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

### Shape of the data

In [177]:
# Using shape attribute of pandas
airbnb_df.shape

(48895, 16)

The **first value** of the tuple obtained will denote to the **number of rows** i.e. **(48895)**in the dataset whereas the **second value** will denote the **number of columns** i.e. **(16)** in the dataset.

### Dataset information

In [178]:
# Details of all the columns present in the dataset
airbnb_df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

The dataset includes attribute-

1. id: (Unique id of listing of all the dataset)
2. name: (Name of the property)
3. host_id: (Unique id for each host in the dataset)
4. host_name: (Name of the host)
5. neighbourhood_group: (Name of the group in the city such as- Bronx, Brooklyn, Manhattan, Queens, Staten Island)
6. neighbourhood: (Specific name of the neighbour where the listing is located)
7. latitude: (Latitude of the location, continuous from 40.49979 to 40.90804)
8. longitude: (Latitude of the location, continuous from -74.24285 to -73.71299)
9. room_type: (Type of room such as- Entire home/apt, Private room, Shared room)
10. price: (Price of the stay per night in USD)
11. minimum_nights: (Minimum number of nights, that a guest will stay at the listing)
12. number_of_reviews: (Total number of reviews the property has recieved)
13. last_review: (Date of the last review recieved by the property)
14. reviews_per_month: (Average number of reviews property recieved per month)
15. calculated_host_listings_count: (Overall number of listing that host has on airbnb)
16. availability_365: (Number of days out of 365 days, the property is available for booking)

### Data Cleaning

In [179]:
# Displaying rows of the dataset using head() function
# By default head() will display 5 rows
# T denotes the transpose of the table (rows becomes columns and columns becomes rows)
airbnb_df.head().T

0  \
id                                                            2539   
name                            Clean & quiet apt home by the park   
host_id                                                       2787   
host_name                                                     John   
neighbourhood_group                                       Brooklyn   
neighbourhood                                           Kensington   
latitude                                                  40.64749   
longitude                                                -73.97237   
room_type                                             Private room   
price                                                          149   
minimum_nights                                                   1   
number_of_reviews                                                9   
last_review                                             2018-10-19   
reviews_per_month                                             0.21   
calculated_host_listings_count                                   6   
availability_365                                               365   

                                                    1  \
id                                               2595   
name                            Skylit Midtown Castle   
host_id                                          2845   
host_name                                    Jennifer   
neighbourhood_group                         Manhattan   
neighbourhood                                 Midtown   
latitude                                     40.75362   
longitude                                   -73.98377   
room_type                             Entire home/apt   
price                                             225   
minimum_nights                                      1   
number_of_reviews                                  45   
last_review                                2019-05-21   
reviews_per_month                                0.38   
calculated_host_listings_count                      2   
availability_365                                  355   

                                                                  2  \
id                                                             3647   
name                            THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                                        4632   
host_name                                                 Elisabeth   
neighbourhood_group                                       Manhattan   
neighbourhood                                                Harlem   
latitude                                                   40.80902   
longitude                                                  -73.9419   
room_type                                              Private room   
price                                                           150   
minimum_nights                                                    3   
number_of_reviews                                                 0   
last_review                                                     NaN   
reviews_per_month                                               NaN   
calculated_host_listings_count                                    1   
availability_365                                                365   

                                                              3  \
id                                                         3831   
name                            Cozy Entire Floor of Brownstone   
host_id                                                    4869   
host_name                                           LisaRoxanne   
neighbourhood_group                                    Brooklyn   
neighbourhood                                      Clinton Hill   
latitude                                               40.68514   
longitude                                             -73.95976   
room_type                                       Entire home/apt   
price                     

In [180]:
# Basic information of the dataset using info() function
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

The info() function will display the total count of non-null values and the data types of different columns.

In [181]:
# Dropping for duplicate values of rows using drop_duplicates() function
airbnb_df = airbnb_df.drop_duplicates()
airbnb_df.count()  #count() function will count the number of rows after dropping duplicates

id                                48895
name                              48879
host_id                           48895
host_name                         48874
neighbourhood_group               48895
neighbourhood                     48895
latitude                          48895
longitude                         48895
room_type                         48895
price                             48895
minimum_nights                    48895
number_of_reviews                 48895
last_review                       38843
reviews_per_month                 38843
calculated_host_listings_count    48895
availability_365                  48895
dtype: int64

The number of rows before dropping duplicates and number of rows after dropping duplicates are same. Hence, there is not any duplicates in the dataset.

In [182]:
# Check for null values in each column
# isnull() function returns value
airbnb_df.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

The columns **name** & **host_name** have very few null values which can handled easily by replacing null values with some values.

***Handling missing data***

In [183]:
# Filling missing values in name & host_name with "No name"
airbnb_df["name"].fillna ('no name', inplace = True)
airbnb_df["host_name"].fillna ("no name", inplace = True)
airbnb_df.isnull().sum()

id                                    0
name                                  0
host_id                               0
host_name                             0
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [184]:
# Concating the columns (number_of_reviews, reviews_per_month and last_review) where number_of_reviews = 0 & reviews_per_month = 0
new_num_of_reviews_df = airbnb_df[airbnb_df ["number_of_reviews"] == 0]
new_reviews_df = airbnb_df[airbnb_df ["reviews_per_month"]==0]
new_df = pd.concat([new_num_of_reviews_df,new_reviews_df], axis = 0)
new_df [["number_of_reviews","reviews_per_month","last_review"]]

number_of_reviews  reviews_per_month last_review
2                      0                NaN         NaN
19                     0                NaN         NaN
26                     0                NaN         NaN
36                     0                NaN         NaN
38                     0                NaN         NaN
...                  ...                ...         ...
48890                  0                NaN         NaN
48891                  0                NaN         NaN
48892                  0                NaN         NaN
48893                  0                NaN         NaN
48894                  0                NaN         NaN

[10052 rows x 3 columns]

From the above table, we can see that there are 10052 rows where the **reviews_per_month** column has null values and **number_of_reviews** = 0. So, we can say that **reviews_per_month** = 0 when **number_of_reviews** = 0. The missing values of **reviews_per_month** can replaced by 0 as there is no reviews for such listing.

In [185]:
# Replacing all the null values in review_per_month column by 0
airbnb_df["reviews_per_month"] = airbnb_df["reviews_per_month"].replace(np.nan, value=0)
airbnb_df.isnull().sum()

id                                    0
name                                  0
host_id                               0
host_name                             0
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                     0
calculated_host_listings_count        0
availability_365                      0
dtype: int64

The **last_review** column can be dropped as it tell us about the last date of the review, but when **number_of_reviews** = 0 the value of **last_review** will always be 0 and it will not be required in the analysis

In [186]:
#Dropping last_review column using drop() function
airbnb_df = airbnb_df.drop(['last_review'], axis=1)
airbnb_df.head().T

0  \
id                                                            2539   
name                            Clean & quiet apt home by the park   
host_id                                                       2787   
host_name                                                     John   
neighbourhood_group                                       Brooklyn   
neighbourhood                                           Kensington   
latitude                                                  40.64749   
longitude                                                -73.97237   
room_type                                             Private room   
price                                                          149   
minimum_nights                                                   1   
number_of_reviews                                                9   
reviews_per_month                                             0.21   
calculated_host_listings_count                                   6   
availability_365                                               365   

                                                    1  \
id                                               2595   
name                            Skylit Midtown Castle   
host_id                                          2845   
host_name                                    Jennifer   
neighbourhood_group                         Manhattan   
neighbourhood                                 Midtown   
latitude                                     40.75362   
longitude                                   -73.98377   
room_type                             Entire home/apt   
price                                             225   
minimum_nights                                      1   
number_of_reviews                                  45   
reviews_per_month                                0.38   
calculated_host_listings_count                      2   
availability_365                                  355   

                                                                  2  \
id                                                             3647   
name                            THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                                        4632   
host_name                                                 Elisabeth   
neighbourhood_group                                       Manhattan   
neighbourhood                                                Harlem   
latitude                                                   40.80902   
longitude                                                  -73.9419   
room_type                                              Private room   
price                                                           150   
minimum_nights                                                    3   
number_of_reviews                                                 0   
reviews_per_month                                               0.0   
calculated_host_listings_count                                    1   
availability_365                                                365   

                                                              3  \
id                                                         3831   
name                            Cozy Entire Floor of Brownstone   
host_id                                                    4869   
host_name                                           LisaRoxanne   
neighbourhood_group                                    Brooklyn   
neighbourhood                                      Clinton Hill   
latitude                                               40.68514   
longitude                                             -73.95976   
room_type                                       Entire home/apt   
price                                                        89   
minimum_nights                                                1   
number_of_reviews                                           270   
reviews_per_month      

**Checking unique values for each variable**

In [188]:
#Check for unique id
airbnb_df ["id"].nunique()

48895

In [189]:
#Check for unique host_id
airbnb_df ["host_id"].nunique()

37457

In [190]:
#Check for unique neighbourhood_group
airbnb_df ["neighbourhood_group"].nunique()

5

In [191]:
#Check for unique neighbourhood
airbnb_df ["neighbourhood"].nunique()

221

In [192]:
#Check for unique room type
airbnb_df ["room_type"].nunique()

3

### Data Visualizing